imap_processing/swapi/l2/swapi_l2.py

```diff
@@ -0,0 +1,115 @@
+"""SWAPI L2 processing module."""
+
+import logging
+
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.swapi.swapi_utils import SWAPIAPID
+
+logger = logging.getLogger(__name__)
+
+
+def process_science(l1_dataset: xr.Dataset, data_version: str) -> xr.Dataset:
+    """
+    Produce science data to L2.
+
+    To process science data to L2, we need to:
+    - convert counts to rates. This is done by dividing the counts by the
+        t_bin time. t_bin is the exposure time per energy bin which is
+        obtained by dividing the time for one complete sweep
+        (12 s, coarse + fine sweep) by the total energy steps (72),
+        i.e., t_bin = 12/72 = 0.167 s. This will be constant.
+
+    - update uncertainty. Calculate new uncertainty value using
+        SWP_PCEM_ERR data from level one and divide by t_bin. Eg.
+            SWP_PCEM_UNC = SWP_PCEM_ERR / t_bin
+        Do the same for SCEM and COIN data.
+
+    Parameters
+    ----------
+    l1_dataset : xarray.Dataset
+        The L1 data input.
+    data_version : str
+        Version of the data product being created.
+
+    Returns
+    -------
+    data : xarray.Dataset
+        Processed data to L2.
+    """
+    # Load the CDF attributes
+    cdf_manager = ImapCdfAttributes()
+    cdf_manager.add_instrument_global_attrs("swapi")
+    cdf_manager.load_variable_attributes("imap_swapi_variable_attrs.yaml")
+
+    t_bin = 0.167  # seconds
+    l2_dataset = l1_dataset.copy()
```

Comment:
I'm somewhat hesitant of this since you're copying a lot of variables and also attributes over as well. You are cleaning it up below, but if we end up changing l1 data later, we may forget to come in here and drop those extra variables and such.

Would it be easy enough to "start fresh" in a sense, and only add to the dataset later rather than drop things? Something like `l2_dataset = xr.Dataset(l1_dataset["epoch"], ...)`